src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToRfc3339Transformer.php

```diff
@@ -32,15 +39,25 @@ public function transform($dateTime)
         }
 
         if ($this->inputTimezone !== $this->outputTimezone) {
-            $dateTime = clone $dateTime;
+            if (!$dateTime instanceof \DateTimeImmutable) {
+                $dateTime = clone $dateTime;
+            }
```

Comment:
This makes even more sense when we consider that a `DateTime` object will be returned by `reverseTransform()` anyway, and actually it sets the output timezone on the input data...

So I guess I should change it in all three, thanks @xabbuh for pointing this.